sparkler-api/src/main/java/edu/usc/irds/sparkler/model/Resource.java

```diff
@@ -58,42 +73,53 @@ public Resource(String url, String group, JobContext sparklerJob, Date lastFetch
 
     @Override
     public String toString() {
-        return String.format("Resource(%s, $s, %s, %s, %s, %s, %s, %s)",
-                id, group, lastFetchedAt, numTries, numFetches, depth, score, status);
+        return String.format("Resource(%s, $s, %d, %s, %s, %s, %s, %s, %s)", id,
+            group,
+            group_id, lastFetchedAt, numTries, numFetches, depth, score, status);
     }
 
 
     public static String resourceId(String url, JobContext job) {
         return String.format("%s-%s", job.getId(), url);
     }
 
+    public void setGroupId() {
+        group_id = String.valueOf(((new Random()).nextInt(GROUP_RANGE) + 1));
```

Comment:
Please see my first comment, it's an simple way of having a "equiprobable" distribution between (1, GROUP_RANGE).